plugins/outputs/kinesis/README.md

```diff
@@ -89,3 +115,26 @@ String is defined using the default Point.String() value and translated to []byt
 #### custom
 
 Custom is a string defined by a number of values in the FormatMetric() function.
+
+### aggregate_metrics
+
+This will make the plugin gather the metrics and send them as blocks of metrics in Kinesis records. The number of put requests depends on a few factors.
+
+1. If a random key is in use then a block for each shard in the stream will be created unless there isn't enough metrics then as many blocks as metrics.
+1. Each record will be 1020kb in size + partition key
+
+### gzip_records
+
+This will make the plugin compress the data using GZip before the data is shipped to Kinesis.
+GZip is slower than snappy but generally fast enough and gives much better compression. Use GZip in most cases.
+
+If both gzip and snappy are true. GZip wins.
```

Comment:
What if i set both false, do i still get aggregated but uncompressed metrics?

Reply:
Yes, that is correct. They would be bigger so you don't get the saving in bandwidth but you do get the saving in how many put requests you make.

Reply:
Nice, this is awesome stuff. Just getting the metrics rolled up into 1 payload would give you a savings.

Reply:
The rollup depends on the partition key type.
random:
If there is more metrics than shards then make as many payloads as available shards and fill the payloads. If they are too large (greater than 1020kb), make 2 x shards and repeat the process. It does this because we want to use all available shards on random partition keys.
If there is less metrics than shards we just create a payload for each metric. Its 1 metric per payload but if the volume is that low, it shouldn't matter.

All others:
Start with 1 and if it doesn't fit into a block of 1020kb, double the number of payloads. compress at the end.

Compression is done as a second step so you get your saving in puts and then further savings in bandwidth. Each payload is compressed after making it 1020kb or less so the payload will always fit in the put request.

As a side note my tests have shown the following:
4 shards available
Put Records (Count) — Sum:
- Original = 22,500 puts per minute
- Aggregated and compressed = 240 puts per minute

Incoming Data (Bytes) — Sum:
- Original = 5.5mb per minute average
- Aggregated and compressed = 470kb per minute average